app/service/userService.go

```diff
@@ -175,3 +176,50 @@ func UpdateBulkUsers(users model.Users) *model.AppResponse {
 	}
 	return utility.GetSuccessResponse(common.MSG_SAVE_SUCCESS)
 }
+
+//AuthenticateUser authenicate give user and returns authenicated user details.
+func AuthenticateUser(user *model.User) *model.AppResponse {
+	if utility.IsValidMobile(user.Mobile) {
+		userRepo := repo.NewUserRepository()
+		loggedUser, err := userRepo.FindUserByMobile(user)
+		if err != nil {
+			return utility.GetErrorResponse(common.MSG_INVALID_CREDENTIALS)
```

Comment:
2 things can be done here:
 Either compare db with mobile n hash(pwd); if nil : invalid credentials else success

Other:
 if this approach is to be followed , then return specific error msgs i.e "mobile not registered" or "invalid pwd"